onward/app/services/GetPopularInTagDCR.scala

```diff
@@ -0,0 +1,50 @@
+package services
+
+import com.gu.contentapi.client.utils.format.{ArticleDesign, NewsPillar, StandardDisplay}
+import common.Edition
+import contentapi.ContentApiClient
+import feed.MostReadAgent
+import layout.ContentCard
+import model.ContentFormat
+import model.dotcomrendering.{OnwardCollectionResponse, Trail}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait GetPopularInTagDCR {
```

Comment:
the existing implementation is embedded [in the `Related` trait, which extends `ConciergeRepository`](https://github.com/guardian/frontend/blob/a9339ca7b260256accbb82ed2c9b43708a0118ac/onward/app/services/repositories.scala#L10), which in turn has some error-handling logic. It's not obvious that this error-handling logic is used for the existing implementation of popular-in-tag, but it's perhaps a Chesterton's fence; would be good if we were clearer on what it is doing in the existing implementation.

Reply:
I'd be very tempted to just add these methods to the controller as `trait`s should be abstractions that are used in multiple places. I can't really see where we might extend this elsewhere, and might be better to not allow that as we currently don't fully understand it.

This might also help with understanding what it does.